sycl/test/fpga_tests/fpga_io_pipes.cpp

```diff
@@ -0,0 +1,131 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUNx: %ACC_RUN_PLACEHOLDER %t.out
+//==------------ fpga_io_pipes.cpp - SYCL FPGA pipes test ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+#include <CL/sycl/intel/fpga_extensions.hpp>
+#include <iostream>
+#include <fstream>
+
+#include "io_pipe_def.h"
+
+// TODO: run is disabled, since no support added in FPGA backend yet. Check
+// implementation correctness from CXX and SYCL languages perspective.
+
+// This test is supposed to be run only on Intel FPGA emulator. Change it when
+// we have more experience with IO pipe feature in SYCL.
+// The emulator creates files (one for I pipe, another for O pipe) with the
+// appropriate naming, where a data flowing through a pipe can be stored.
+// So in the test we need to create these files and use them appropriately.
+// The name is taken as IO pipe ID.
+const size_t InputData = 42;
+const std::string InputFileName = "0.txt";
+const std::string OutputFileName = "1.txt";
+
+void fillInput(const std::string &filename) {
+  std::ofstream Input(filename);
+  if (Input.is_open()) {
+    Input << InputData;
+    Input.close();
+  }
+}
+
+int validateOutput(const std::string &filename) {
+  std::ifstream Output(filename);
+  std::string Line;
+  std::vector<size_t> Result;
+  if (Output.is_open()) {
+    while (std::getline(Output, Line))
+      Result.push_back(stoi(Line));
```

Comment:
You would have written a char it would have been quite more understandable.
And since you are just reading 1 line, just removing the loop it enough.

Reply:
I disagree. I can imagine a backend having a bug, that makes it incorrectly interprets `capacity` of a pipe and therefore multiple values/lines will be written in the output file.

Reply:
So, I'd like to leave it as is for sanity-check.

Reply:
OK, I understand your point. But then what about explaining with some comments what you expect to test?
I love comments! :-)

Reply:
Added. Sorry for force-pushing one more time.